Add helm chart for KServe

yuzisun opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://www.kubeflow.org/docs/about/contributing/ and developer guide https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:
This PR adds the helm chart for installing kserve and modelmesh

## Install KServe
`helm install kserve manifests/charts -n kserve`

## Upgrade KServe
 `kubectl apply -f manifests/charts/crds`
`helm upgrade kserve manifests/charts -n kserve`

## Uninstall KServe
`helm uninstall kserve`


**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

yuzisun committed:
Add webhook manifests

yuzisun committed:
Add certificate

yuzisun committed:
Add param values for kserve chart

yuzisun committed:
Fix webhook

yuzisun committed:
Add modelmesh components

yuzisun committed:
Add modelmesh configs

pvaneck commented on manifests/charts/templates/deployment.yaml:
```diff
@@ -0,0 +1,150 @@
+              fieldPath: metadata.name
+        - name: ETCD_SECRET_NAME
+          value: model-serving-etcd
+        image: {{ .Values.kserve.modelmesh-controller.image }}
```
```suggestion
        image: {{ .Values.kserve.modelmesh.controller.image }}
```

pvaneck commented on manifests/charts/templates/rolebinding.yaml:
```diff
@@ -0,0 +1,60 @@
+    app.kubernetes.io/instance: modelmesh-controller
+    app.kubernetes.io/managed-by: modelmesh-controller
+    app.kubernetes.io/name: modelmesh-controller
+  name: controller-leader-election-rolebinding
```
```suggestion
  name: modelmesh-controller-leader-election-rolebinding
```

pvaneck commented on manifests/charts/templates/rolebinding.yaml:
```diff
@@ -0,0 +1,60 @@
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: controller-leader-election-role
```
```suggestion
  name: modelmesh-controller-leader-election-role
```

pvaneck commented on manifests/charts/templates/rolebinding.yaml:
```diff
@@ -0,0 +1,60 @@
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: controller-role
```
```suggestion
  name: modelmesh-controller-role
```

pvaneck commented on manifests/charts/values.yaml:
```diff
@@ -0,0 +1,87 @@
+      modelmeshImage: kserve/modelmesh
+      modelmeshImageTag: v0.7.0
+      modelmeshRuntimeAdapterImage: kserve/modelmesh-runtime-adapter
+      modelmeshRuntimeAdapterTag: v0.7.0
```
```suggestion
      modelmeshRuntimeAdapterImageTag: v0.7.0
      restProxyImage: kserve/rest-proxy
      restProxyImageTag: v0.1.1
```

pvaneck reviewed (changes requested):
Thanks, @yuzisun. For the modelmesh related yaml, I had to make a few changes to get things to work which I added inline. After that, the deployment using helm worked well, and I as able to deploy a predictor (after configuring etcd).

yuzisun committed:
Update template variables

Co-authored-by: Paul Van Eck <pvaneck@us.ibm.com>

yuzisun committed:
Apply suggestions from code review

Co-authored-by: Paul Van Eck <pvaneck@us.ibm.com>

yuzisun commented:
> Thanks, @yuzisun. For the modelmesh related yaml, I had to make a few changes to get things to work which I added inline. After that, the deployment using helm worked well, and I as able to deploy a predictor (after configuring etcd).

Thanks @pvaneck ! Forgot to run another test after adding MM template variables, updated the PR now.

pvaneck reviewed (commented):
/lgtm
